internal/cache/file/downloader/job.go

```diff
@@ -166,6 +166,36 @@ func (job *Job) failWhileDownloading(downloadErr error) {
 	job.mu.Unlock()
 }
 
+// updateFileInfoCache updates the file info cache with latest offset downloaded
+// by job. Returns error in case of failure.
+//
+// Not concurrency safe and requires LOCK(job.mu)
+func (job *Job) updateFileInfoCache() (err error) {
+	fileInfoKey := data.FileInfoKey{
```

Comment:
As discussed let's make the fileInfoKeyName as a member variable of Job, so that we don't need to calculate every time.

Reply:
Even if we make fileInfoKeyName as member of Job class, we will have to create fileInfoKey in this function every time to insert into the fileInfoCache.
So either we have to keep both fileInfoKey and fileInfoKeyName as members or keep fileInfoKey as member and get fileInfoKeyName everytime in this function. 
To keep things simple, we can go ahead with the current approach ?

Reply:
Okay, for now we can skip. We will take it separately to optimize fileInfoKey creation part globally.